java/pgjdbc_using_dsql_connector/src/main/java/com/amazon/dsql/samples/CustomCredentialsProviderExample.java

```diff
@@ -0,0 +1,191 @@
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sts.StsClient;
+import software.amazon.awssdk.services.sts.auth.StsAssumeRoleCredentialsProvider;
+import software.amazon.awssdk.services.sts.model.AssumeRoleRequest;
+import software.amazon.dsql.jdbc.AuroraDsqlCredentialsManager;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+/**
+ * Custom AWS Credentials Provider example using Aurora DSQL JDBC Connector.
+ * This example demonstrates how to use different AWS credential providers
+ * with the Aurora DSQL JDBC Connector.
+ */
+public class CustomCredentialsProviderExample {
+    
+    // Read configuration from environment variables
+    private static final String CLUSTER_ENDPOINT = System.getenv("CLUSTER_ENDPOINT");
+    private static final String CLUSTER_USER = System.getenv("CLUSTER_USER");
+    private static final String PROFILE = System.getenv("PROFILE");
+    private static final String ROLE_ARN = System.getenv("ROLE_ARN");
+    
+    public static void main(String[] args) {
+        System.out.println("Aurora DSQL JDBC Connector - Custom Credentials Provider Example");
+        
+        // Verify environment variables are set
+        assert CLUSTER_ENDPOINT != null : "CLUSTER_ENDPOINT environment variable is not set";
+        assert CLUSTER_USER != null : "CLUSTER_USER environment variable is not set";
+        
+        // Example 1: Using Profile Credentials Provider
+        connectWithProfileCredentials();
+        
+        // Example 2: Using STS Assume Role Credentials Provider
+        connectWithAssumeRoleCredentials();
+    }
+    
+    /**
+     * Connect using a specific AWS profile
+     */
+    private static void connectWithProfileCredentials() {
+        System.out.println("\n=== Example 1: Profile Credentials Provider ===");
+        
+        try {
+            // Verify environment variables are set
+            assert CLUSTER_ENDPOINT != null : "CLUSTER_ENDPOINT environment variable is not set";
+            assert CLUSTER_USER != null : "CLUSTER_USER environment variable is not set";
+            
+            // Create profile credentials provider
+            AwsCredentialsProvider profileCredentials = ProfileCredentialsProvider.builder()
+                    .profileName(PROFILE != null ? PROFILE : "default") // Use environment variable or default
+                    .build();
+            
+            // Set the custom credentials provider
+            AuroraDsqlCredentialsManager.setProvider(profileCredentials);
+            
+            // Connect to Aurora DSQL using URL with profile parameter
+            String url = "jdbc:aws-dsql:postgresql://" + CLUSTER_ENDPOINT + "/postgres?user=" + CLUSTER_USER;
+            if (PROFILE != null && !PROFILE.isEmpty()) {
+                url += "&profile=" + PROFILE;
```

Comment:
Is the profile still needed as a parameter if it is configured using the `profileCredentials` above?